$OpenBSD: patch-taglib_mpeg_id3v2_id3v2frame_cpp,v 1.1 2012/10/16 13:06:13 dcoppa Exp $

Fix saving of PropertyMap comments without description into ID3v2
tags (upstream git commit c8994ede3fdc8569a59b3482acbc797186e72d00)

--- taglib/mpeg/id3v2/id3v2frame.cpp.orig	Tue Oct 16 13:47:07 2012
+++ taglib/mpeg/id3v2/id3v2frame.cpp	Tue Oct 16 13:49:16 2012
@@ -120,7 +120,7 @@ Frame *Frame::createTextualFrame(const String &key, co
       TextIdentificationFrame *frame = new TextIdentificationFrame(frameID, String::UTF8);
       frame->setText(values);
       return frame;
-    } else if(values.size() == 1){  // URL frame (not WXXX); support only one value
+    } else if((frameID[0] == 'W') && (values.size() == 1)){  // URL frame (not WXXX); support only one value
         UrlLinkFrame* frame = new UrlLinkFrame(frameID);
         frame->setUrl(values.front());
         return frame;
@@ -144,7 +144,9 @@ Frame *Frame::createTextualFrame(const String &key, co
   // -COMMENT: depending on the number of values, use COMM or TXXX (with description=COMMENT)
   if((key == "COMMENT" || key.startsWith(commentPrefix)) && values.size() == 1){
     CommentsFrame *frame = new CommentsFrame(String::UTF8);
-    frame->setDescription(key == "COMMENT" ? key : key.substr(commentPrefix.size()));
+    if (key != "COMMENT"){
+      frame->setDescription(key.substr(commentPrefix.size()));
+    }
     frame->setText(values.front());
     return frame;
   }
